src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.ShoppingCartDao;
+import com.dev.cinema.exception.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import org.hibernate.Hibernate;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot add Shopping cart "
+                    + shoppingCart + "into DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery("from ShoppingCart "
+                    + "where id = :userId", ShoppingCart.class);
```

Comment:
It would be nice to use JOIN FETCH for tickets and user here. You can also fetch the user directly without referring to the user id, e.g.
```suggestion
            Query<ShoppingCart> query = session.createQuery(""FROM ShoppingCart sc "
                    + "JOIN FETCH sc.user u "
                    + "LEFT JOIN FETCH sc.tickets t "
                    + "WHERE sc.user = :user", ShoppingCart.class);
```

Reply:
done